File: app/src/main/java/za/co/xisystems/itis_rrm/ui/mainview/corrections/CorrectionsFragment.java
package za.co.xisystems.itis_rrm.ui.mainview.corrections;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import za.co.xisystems.itis_rrm.R;

public class CorrectionsFragment extends Fragment {

    private CorrectionsViewModel correctionsViewModel;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        correctionsViewModel =
                ViewModelProviders.of(this).get(CorrectionsViewModel.class);
        View root = inflater.inflate(R.layout.fragment_correction, container, false);
//        final TextView textView = root.findViewById(R.id.text_tools);
        correctionsViewModel.getText().observe(this, new Observer<String>() {
            @Override
            public void onChanged(@Nullable String s) {
//                textView.setText(s);
            }
        });
        return root;
    }
}

File: app/src/main/java/za/co/xisystems/itis_rrm/ui/mainview/corrections/CorrectionsFragment.kt
@file:Suppress("SpellCheckingInspection")

package za.co.xisystems.itis_rrm.ui.mainview.corrections

import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.ViewModelProvider
import org.kodein.di.DIAware
import org.kodein.di.android.x.closestDI
import org.kodein.di.instance
import za.co.xisystems.itis_rrm.R
import za.co.xisystems.itis_rrm.base.BaseFragment

/**
 * Created by Francis Mahlava on 03,October,2019
 */

class CorrectionsFragment : BaseFragment(), DIAware {
    //
    override val di by closestDI()
    private lateinit var correctionsViewModel: CorrectionsViewModel
    private val factory: CorrectionsViewModelFactory by instance()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        correctionsViewModel =
            ViewModelProvider(this.requireActivity(), factory).get(CorrectionsViewModel::class.java)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        if (view != null) {
            val parent = requireView().parent as ViewGroup
            parent.removeAllViews()
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_correction, container, false)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // no options menu
        return false
    }
}
